ansible_collections/arista/avd/roles/eos_designs/templates/base/base.py

```diff
@@ -0,0 +1,148 @@
+    def _mgmt_destination_networks(self):
+        return get(self._host_vars, "mgmt_destination_network")
+
+    @cached_property
+    def _switch_data(self):
+        return get(self._host_vars, 'switch', {})
+
+    @cached_property
+    def _switch_bgp_as(self):
+        return get(self._switch_data,'bgp_as')
+
+    @cached_property
+    def _switch_router_id(self):
+        return get(self._switch_data, 'router_id')
+
+    @cached_property
+    def _switch_bgp_defaults(self):
+        return get(self._switch_data, 'bgp_defaults')
+
+    @cached_property
+    def _bgp_maximum_paths(self):
+        return get(self._host_vars, "bgp_maximum_paths")
+
+    @cached_property
+    def _bgp_ecmp(self):
+        return get(self._host_vars, "bgp_ecmp")
+
+    @cached_property
+    def _load_interval_default(self):
+        return get(self._host_vars, "load_interval_default")
+
+    @cached_property
+    def _hardware_counters(self):
+        return get(self._host_vars, "hardware_counters")
+
+    @cached_property
+    def _hardware_counters_features(self):
+        return get(self._hardware_counters, 'features')
+
+    @cached_property
+    def _internal_vlan_order(self):
+        return get(self._host_vars, 'internal_vlan_order')
+
+    @cached_property
+    def _allocation(self):
+        return get(self._internal_vlan_order, 'allocation')
+
+    @cached_property
+    def _range(self):
+        return get(self._internal_vlan_order, 'range')
+
+    @cached_property
+    def _beginning(self):
+        return get(self._range, 'beginning')
+
+    @cached_property
+    def _ending(self):
+        return get(self._range, 'ending')
+
+    @cached_property
+    def static_routes(self):
+        static_routes = []
+        if "mgmt_destination_networks" in self._host_vars and len(self._mgmt_destination_networks) > 0:
+            for mgmt_destination_network in self._mgmt_destination_networks:
+                static_routes.append({'vrf': self._mgmt_interface_vrf,
+                                      'destination_address_prefix': mgmt_destination_network,
+                                      'gateway': self._mgmt_gateway})
+        else:
+            static_routes.append({'vrf': self._mgmt_interface_vrf,
+                                  'destination_address_prefix': '0.0.0.0/0',
+                                  'gateway': self._mgmt_gateway})
+        return static_routes
+
+    @cached_property
+    def router_bgp(self):
+        if "bgp_as" in self._switch_data:
+            router_bgp = {'as': self._switch_bgp_as,
+                          'router_id': self._switch_router_id,
+                          'bgp_defaults': []}
+            for bgp_default in self._switch_bgp_defaults:
+                router_bgp['bgp_defaults'].append(bgp_default)
+
+            if "bgp_maximum_paths" in self._host_vars and "bgp_ecmp" in self._host_vars:
+                router_bgp['bgp_defaults'].append("maximum-paths " + str(self._bgp_maximum_paths)
```

Comment:
please use f-strings everywhere for variable insertion in strings. It makes it more readable.